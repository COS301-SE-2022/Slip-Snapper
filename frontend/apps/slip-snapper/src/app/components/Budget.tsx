import {
    IonAlert,
    IonInput,
    IonItem,
    IonProgressBar,
} from "@ionic/react";
import { useState } from "react";


function Budget() {
    const [food, setFood] = useState(false);
    const [fashion, setFashion] = useState(false);
    const [elec, setElec] = useState(false);
    const [houseHold, setHouseHold] = useState(false);
    const [other, setOther] = useState(false);

    const [amounts, setAmounts] = useState<string[]>(["100", "200", "300", "400", "500"]);

    return (
        <div>
            <IonItem onClick={() => setFood(true)} id="foodBudget" className="center-items" color="tertiary">
                <IonInput readonly value={"Food: R"+amounts[0]}></IonInput>
                <IonProgressBar id='foodBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={food}
                onDidDismiss={() => setFood(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food',placeholder: 'Insert Food Budget'},]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food,0);
                            isExceeded(parseFloat(alertData.food),"foodBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setFashion(true)} id="fashionBudget" className="center-items" color="tertiary">
                <IonInput readonly value={"Fashion: R" + amounts[1]}></IonInput>
                <IonProgressBar id='fashionBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={fashion}
                onDidDismiss={() => setFashion(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Fashion Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 1);
                            isExceeded(parseFloat(alertData.food), "fashionBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setElec(true)} id="electronicsBudget" className="center-items" color="tertiary">
                <IonInput readonly value={"Electronics: R" + amounts[2]}></IonInput>
                <IonProgressBar id='elecBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={elec}
                onDidDismiss={() => setElec(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Electronics Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 2);
                            isExceeded(parseFloat(alertData.food), "elecBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setHouseHold(true)} id="weekly-budget" className="center-items" color="tertiary">
                <IonInput readonly value={"Household: R" + amounts[3]}></IonInput>
                <IonProgressBar id='houseBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={houseHold}
                onDidDismiss={() => setHouseHold(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Fashion Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 3);
                            isExceeded(parseFloat(alertData.food), "houseBar")
                        }
                    }]}></IonAlert>

            <IonItem onClick={() => setOther(true)} id="weekly-budget" className="center-items" color="tertiary">
                <IonInput readonly value={"Other: R" + amounts[4]}></IonInput>
                <IonProgressBar id='otherBar' class='progressBar'></IonProgressBar><br />
            </IonItem>
            <IonAlert
                isOpen={other}
                onDidDismiss={() => setOther(false)}
                header={'Change Budget'}
                inputs={[
                    { name: 'food', placeholder: 'Insert Fashion Budget' },]}
                buttons={[
                    {
                        text: 'Apply',
                        handler: (alertData) => {
                            applyToBudget(alertData.food, 4);
                            isExceeded(parseFloat(alertData.food), "otherBar")
                        }
                    }]}></IonAlert>
        </div>
    );

    function applyToBudget(newValue: string ,pos:number) {
        const val = parseFloat(newValue)

        if (!isNaN(val)) {
            amounts[pos]=val.toString()
            setAmounts(amounts)
        }
    }

  
    function isExceeded(budget:number,barID:string) {
        const totalFoodSpent = 88;
        const withinBudget = totalFoodSpent / budget

        if (totalFoodSpent >= budget && !isNaN(budget)) {
            document.getElementById(barID)?.setAttribute("color", "danger")
        }
        else if (totalFoodSpent >= budget / 2 && !isNaN(budget)) {
            document.getElementById(barID)?.setAttribute("color", "warning")
        }
        else if (!isNaN(budget)) {
            document.getElementById(barID)?.setAttribute("color", "success")
        }

        document.getElementById(barID)?.setAttribute("value", withinBudget.toString())

        if (budget === 0) {
            document.getElementById(barID)?.setAttribute("value", "0")
            document.getElementById(barID)?.setAttribute("color", "success")
        }

      
    }
}
export default Budget;