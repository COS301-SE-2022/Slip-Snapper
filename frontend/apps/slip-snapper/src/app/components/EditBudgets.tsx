import {
  IonButton,
  IonButtons,
  IonItem,
  IonLabel,
  IonModal,
  IonHeader,
  IonToolbar,
  IonTitle,
  IonContent,
  IonCheckbox,
  IonList
} from "@ionic/react";
import React, { useState } from 'react';

export const EditBudgets = () => {
  const [isOpen, setIsOpen] = useState(false);

  const [food, setFood] = useState(false);
  const [fashion, setFashion] = useState(false);
  const [electronics, setElectronics] = useState(false);
  const [household, setHousehold] = useState(false);
  const [other, setOther] = useState(false);

  const categoryStates = [
    { name: 'Food', isChecked: food, id: "foodBudget" },
    { name: 'Fashion', isChecked: fashion, id: "fashionBudget" },
    { name: 'Electronics', isChecked: electronics, id: "electronicsBudget" },
    { name: 'Household', isChecked: household, id: "houseBudget" },
    { name: 'Other', isChecked: other, id: "otherBudget" }
  ];
  hideData(categoryStates)

  return (
    <IonItem color="primary">
      <IonButton fill="solid" color="secondary" onClick={() => setIsOpen(true)}>
        <IonLabel>Add/Remove</IonLabel>
      </IonButton>
      <IonModal isOpen={isOpen} onDidDismiss={() => setIsOpen(false)}>
        <IonHeader>
          <IonToolbar color="primary">
            <IonTitle>Add/Remove Budgets</IonTitle>
            <IonButtons slot="end">
              <IonButton onClick={() => {
                setIsOpen(false);
                hideData(categoryStates)
              }}>Confirm</IonButton>
            </IonButtons>
          </IonToolbar>
        </IonHeader>
        <IonContent>
          <IonList>
            <IonItem>
              <IonLabel>Food</IonLabel>
              <IonCheckbox slot="end" checked={food} onIonChange={e => setFood(e.detail.checked)} />
            </IonItem>

            <IonItem>
              <IonLabel>Fashion</IonLabel>
              <IonCheckbox slot="end" checked={fashion} onIonChange={e => setFashion(e.detail.checked)} />
            </IonItem>

            <IonItem>
              <IonLabel>Electronics</IonLabel>
              <IonCheckbox slot="end" checked={electronics} onIonChange={e => setElectronics(e.detail.checked)} />
            </IonItem>

            <IonItem>
              <IonLabel>Household</IonLabel>
              <IonCheckbox slot="end" checked={household} onIonChange={e => setHousehold(e.detail.checked)} />
            </IonItem>

            <IonItem>
              <IonLabel>Other</IonLabel>
              <IonCheckbox slot="end" checked={other} onIonChange={e => setOther(e.detail.checked)} />
            </IonItem>
          </IonList>
        </IonContent>
      </IonModal>
    </IonItem>
  );
};


export function hideData(categoryStates:any) {
  for (let i = 0; i < categoryStates.length; i++) {
    const temp = document.getElementById(categoryStates[i].id)
    if (categoryStates[i].isChecked === true) {

      if (temp !== null)
        temp.style.display = ""
    }
    else {
      if (temp !== null)
        temp.style.display = "none"
    }

  }
}


