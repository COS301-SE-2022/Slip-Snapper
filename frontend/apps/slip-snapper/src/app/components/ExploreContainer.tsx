import React from 'react';
import './ExploreContainer.css';

const ExploreContainer = () => {
  return (
    <div className="container">
      <h1>Welcome To Slip Snapper</h1>
      <strong>Got a Reciept Slip ready to Snap?</strong>
      <p>
          
      </p>
    </div>
  );
};

export default ExploreContainer;
