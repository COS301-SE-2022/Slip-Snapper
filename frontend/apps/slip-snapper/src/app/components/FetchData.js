import React from "react";
import { IonButton } from '@ionic/react';
import '../pages/FetchData.css';
class FetchData extends React.Component {
   
    constructor(props) {
        super(props);
   
        this.state = {
            items: [],
            DataisLoaded: false
        };
    }
   
    componentDidMount() {
        fetch("http://localhost:1234/items?user=1", {
                method: 'get',
                headers: {
                    'Content-Type': 'application/json'
                }
            })
            .then((res) => res.json())
            .then(
                (json) => {
                this.setState({
                    items: json,
                    DataisLoaded: true
                });
            },
                (error) => {
                this.setState({
                    DataisLoaded: true,
                    error
                });
            })
    }
    render() {
        const { DataisLoaded, items } = this.state;
        if (!DataisLoaded) return <div>
            <h1> Items Loading.... </h1> </div> ;
        return (
        <div className = "App">
            <h1> Items </h1>  {
                items.map((item) => ( 
                <div className="div" id={item.id} contentEditable="false" suppressContentEditableWarning={true}>
                    <h2>Item { item.id }</h2>
                    <ul key = { item.id } className="no-bullets">
                        <li>Item Name: { item.item_name }</li>
                        <li>Quantity: { item.quantity }</li>
                        <li>Price: { item.price }</li>
                        <li>Type: { item.type }</li>
                        <li>Date: { item.date }</li>
                        <li>Location: { item.location }</li>
                    </ul>
                    <IonButton id={item.id+"b"} fill="outline" color="secondary" onClick={()=>toggleEditable(item.id)}>Edit</IonButton>
                </div>
                ))
            }
        </div>
    );

    function toggleEditable(id) {
        var editDiv = document.getElementById(id);
        var btn = document.getElementById(id+"b");
        var editedText = document.getElementById(id).innerText;
        if (editDiv.contentEditable === "true") {
            editDiv.contentEditable = "false";
            btn.color = "secondary";
            console.log(editedText);
        } else {
            editDiv.contentEditable = "true";
            btn.color = "tertiary";
        }
    }
}
}
   
export default FetchData;