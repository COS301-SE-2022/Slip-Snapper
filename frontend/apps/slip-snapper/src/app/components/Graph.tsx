import { Bar } from "react-chartjs-2";

type Props = {
    graphData: any;
}
function Graph({ graphData }: Props) {

    const amounts = []
    const frequency = []
    for(let i = 0 ; i<graphData.amounts.length;i++)
    {
        amounts[i] = graphData.amounts[i].itemPrice
        frequency[i] = graphData.occurances[i].id
    }


     const graphSettings = {
        responsive: true,
        barThickness: 43,
        borderWidth: 1,
        borderRadius: 10,
        
        plugins: {
            legend: {
                position: 'bottom' as const,
            },
            title: {
                display: true,
                text: graphData.itemName+' prices across various stores',
            },
        },
    };

     const graphStats = {
         labels: graphData.stores,
        datasets: [
            {
                label: 'Price (R)',
                data: amounts,
                backgroundColor: "rgb(39, 165, 146, 0.8)",
                width: 8,
                hoverBackgroundColor: "rgb(47, 198, 176, 0.8)",
            },
            {
                label: 'Frequency',
                data: frequency,
                backgroundColor: 'rgb(156, 39, 176, 0.8)',
                width: 8,
                hoverBackgroundColor: "rgb(187, 54, 211, 0.8)",
                hidden: true,
            }
        ],
    };

    return (
        <Bar className="graph" options={graphSettings} data={graphStats} />
    );
}

export default Graph;