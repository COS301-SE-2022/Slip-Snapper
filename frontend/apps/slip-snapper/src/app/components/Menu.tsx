import {
    IonMenu,
    IonHeader,
    IonToolbar,
    IonTitle,
    IonContent,
    IonList,
    IonMenuToggle,
    IonItem,
    IonLabel,
    IonIcon
  } from "@ionic/react";
  import React from "react";
  import {home, person, documents} from 'ionicons/icons';

  
  export const Menu = () => {
    return (
      <IonMenu side="end" contentId="main">
        <IonHeader>
          <IonToolbar color="light">
            <IonTitle>Menu</IonTitle>
          </IonToolbar>
        </IonHeader>
        <IonContent>
          <IonList>
            <IonMenuToggle auto-hide="false">
              <IonItem button routerLink={"/home"} routerDirection="none">
                <IonLabel>Home</IonLabel>
                <IonIcon src={home}/>
              </IonItem>
            </IonMenuToggle>
            <IonMenuToggle auto-hide="false">
              <IonItem button routerLink={"/viewreports"} routerDirection="none">
                <IonLabel>Reports</IonLabel>
                <IonIcon src={documents}/>
              </IonItem>
            </IonMenuToggle>
            <IonMenuToggle auto-hide="false">
              <IonItem button routerLink={"/profile"} routerDirection="none">
                <IonLabel>Profile</IonLabel>
                <IonIcon src={person}/>
              </IonItem>
            </IonMenuToggle>
          </IonList>
        </IonContent>
      </IonMenu>
    );
  };