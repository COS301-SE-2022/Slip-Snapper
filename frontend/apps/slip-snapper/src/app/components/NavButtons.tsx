import { IonButton, IonMenuButton } from "@ionic/react";
import React, { useEffect } from "react";

export const NavButtons = () => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const [mQuery, setMQuery] = React.useState<any>({
    matches: window.innerWidth > 768 ? true : false,
  });

  useEffect(() => {
    const mediaQuery = window.matchMedia("(min-width: 768px)");
    mediaQuery.addListener(setMQuery);

    return () => mediaQuery.removeListener(setMQuery);
  }, []);

  console.log(mQuery.matches);

  return (
    <div>
      {mQuery && !mQuery.matches ? (
        <IonMenuButton />
      ) : (
        <>
          <IonButton routerLink={"/home"}>Home</IonButton>
          <IonButton routerLink={"/viewreports"}>Reports</IonButton>
          <IonButton routerLink={"/profile"}>Profile</IonButton>
        </>
      )}
    </div>
  );
};