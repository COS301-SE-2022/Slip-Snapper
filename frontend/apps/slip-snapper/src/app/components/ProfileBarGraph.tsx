import { Bar } from "react-chartjs-2";

type Props = {
    graphData: any;
}
function ProfileBarGraph({ graphData }: Props) {
    const graphSettings = {
        responsive: true,
        barThickness: 43,
        borderWidth: 1,
        borderRadius: 10,
        color: 'white',


        plugins: {
            legend: {
                position: 'bottom' as const,
            },
        },

        scales: {
            yAxes: {
                ticks: {
                    beginAtZero: true,
                    color: 'white',
                    fontSize: 12,
                }
            },
            xAxes: {
                ticks: {
                    beginAtZero: true,
                    color: 'white',
                    fontSize: 12,
                }
            },
        },
    };

    const graphStats = {
   
        datasets: [
            {
                label: 'Expenditure (R)',
                data: graphData,
                backgroundColor: "rgb(39, 165, 146, 0.8)",
                width: 8,
                hoverBackgroundColor: "rgb(47, 198, 176, 0.8)",
                
            }
        ],
    };
    return (
        <Bar className="graph" options={graphSettings} data={graphStats} />
    );
}

export default ProfileBarGraph;