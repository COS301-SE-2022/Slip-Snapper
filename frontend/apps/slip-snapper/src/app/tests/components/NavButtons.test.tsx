import renderer from 'react-test-renderer';
import { NavButtons } from "../../components/NavButtons";

const renderTree = (tree: JSX.Element) => renderer.create(tree);
describe('<NavButtons>', () => {
  it('should render component', () => {
    expect(renderTree(<NavButtons 
    />).toJSON()).toMatchSnapshot();
  });
  
});