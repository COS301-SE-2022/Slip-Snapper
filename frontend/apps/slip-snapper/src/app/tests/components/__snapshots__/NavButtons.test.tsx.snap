// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavButtons> should render component 1`] = `
<div>
  <ion-button
    onClick={[Function]}
  >
    Home
  </ion-button>
  <ion-button
    onClick={[Function]}
  >
    Reports
  </ion-button>
  <ion-button
    onClick={[Function]}
  >
    Profile
  </ion-button>
  <ion-button
    onClick={[Function]}
  >
    Receipts
  </ion-button>
</div>
`;
